### DISCUSSION TITLE
Way to integrate ticker with declarative UI renderers?

### DISCUSSION THREAD

**Discussion by lmcgartland** [pixijs/pixijs#7430] [Q&A]
I'm interested in tweaking the performance of Pixi for use with a declarative UI framework like React, Vue, or Svelte. These frameworks have their own render cycle for updating the DOM. Fortunately, there are already Pixi bindings for use with them. What I'm interested in is getting the Pixi Ticker to align with the render events with these frameworks. When using Pixi with the Svelte bindings for example, it takes 1 frame for Pixi to catch up with the state updates made on the Svelte side (I'm assuming this is a common problem shared by all the frameworks). Would there be a way to link the ticker to render cycle of another UI framework?

**Comment by bigtimebuddy** [comment]
Good question!

There are a few things that will be helpful for you. 

First if you're using Application, the easiest thing to do is stop the ticker and manually render yourself. For instance:

```js
const app = new PIXI.Application({ autoStart: false });
app.render(); // <- manually render within your own update loop
```

Or if you're not using Application and creating a vanilla Renderer, probably something like this:
```js
const renderer = new PIXI.autoDetectRenderer();
const stage = new PIXI.Container();
renderer.render(stage); // manually render without a Ticker
```

There are two other Tickers that you should be aware of, however, these are the `system` and `shared` Tickers. These are global tickers that are used for specific purposes. Please [consult the documentation](https://pixijs.download/dev/docs/PIXI.Ticker.html) for more information about these. If you like, you can stop these and manually call them later:

```js
// Stop both tickers from running automatically
PIXI.Ticker.shared.stop();
PIXI.Ticker.system.stop();

// if using AnimatedSprite or video, otherwise ignore
PIXI.Ticker.shared.update();

// If using interaction or renderer.prepare, otherwise ignore
PIXI.Ticker.system.update();
```
